File: remove_duplicates_from_linked_list/remove_duplicates_from_linked_list.go
package main

import "fmt"

type LinkedList struct {
    value int
    next  *LinkedList
}

func removeDuplicatesFromLinkedList(linkedList *LinkedList) *LinkedList {
    currentNode := linkedList
    for currentNode != nil {
        if currentNode.next != nil && currentNode.value == currentNode.next.value {
            currentNode.next = currentNode.next.next
        } else {
            currentNode = currentNode.next
        }
    }
    return linkedList
}

func main() {
    linkedList := &LinkedList{value: 1}
    linkedList.next = &LinkedList{value: 1}
    linkedList.next.next = &LinkedList{value: 2}

    linkedList = removeDuplicatesFromLinkedList(linkedList)

    fmt.Println(linkedList.value)        
    fmt.Println(linkedList.next.value)  
}

File: remove_duplicates_from_linked_list/remove_duplicates_from_linked_list.ts
class LinkedList {
    value: number;
    next: LinkedList | null;

    constructor(value: number) {
        this.value = value;
        this.next = null;
    }

    static removeDuplicatesFromLinkedList(linkedList: LinkedList | null): LinkedList | null {
        let currentNode = linkedList;
        while (currentNode !== null) {
            const nextNode = currentNode.next;
            if (nextNode !== null && currentNode.value === nextNode.value) {
                currentNode.next = nextNode.next;
            } else {
                currentNode = currentNode.next;
            }
        }
        return linkedList;
    }
}

const linkedList = new LinkedList(1);
linkedList.next = new LinkedList(1);
linkedList.next.next = new LinkedList(2);

const updatedList = LinkedList.removeDuplicatesFromLinkedList(linkedList);
console.log(updatedList); 
console.log(updatedList?.value); 
console.log(updatedList?.next?.value);
